src/bundle/commitments.rs

```diff
@@ -19,6 +27,40 @@ fn hasher(personal: &[u8; 16]) -> State {
     Params::new().hash_length(32).personal(personal).to_state()
 }
 
+// FIXME: Consider not using a separate OrchardHash trait and instead move update_hash_with_burn to
+// the OrchardDomain or OrchardFlavour trait.
```

Comment:
yes, that is too much for one conditional update. let's move it to OrchardDomain or OrchardFlavour

Reply:
Done: used OrchardFlavor instead of OrchardDomainCommon+OrchardCircuit+OrchardHash and renamed the corresponding generic arg from D to FL.